### PR TITLE
[#1782] Implement Probes for recommendation system

### PR DESCRIPTION
**Pull request by danielkryska**
- Fetch recommended services by search state 
  - Send request on search, filters change
- Record user actions
  - Add attribute tags to HTML elements
  - Add event listener on page load and turbo::before

### CODE REVIEW

**Comment by bwilk** [comment]
During the review check whether we are able to operate without recommender system in production

File: app/controllers/probation_controller.rb [hunk 1]
```diff
@@ -0,0 +1,40 @@
+# frozen_string_literal: true
+
+class ProbationController < ApplicationController
```
Comment (michal-szostak) [inline app/controllers/probation_controller.rb:3]: I'm not sure about the name of the controller - I'd rather change it to ProbeController...

File: app/javascript/app/probes.js [hunk 2]
```diff
@@ -0,0 +1,67 @@
+import Rails from '@rails/ujs'
+
+export default function initProbes(scope = window) {
+    window.document.querySelectorAll("[data-probe]")
+        .forEach(function(element) {
+            element.addEventListener('click', async function () {
+                const visit_time = new Date().getTime();
+                const is_ordered = scope.location.pathname.includes("summary")
```
Comment (michal-szostak) [inline app/javascript/app/probes.js:8]: This is a minor, may be omitted - but It's better to have consistent style with string literals in code - I'd advise to pick `'<string>'`

**Review by michal-szostak — APPROVED** [review]
Two minors, but I strongly recommend fixing the naming scheme.
I'd like to have @JanKapala to checkout this build and test it manually with recommender instance

**Comment by Marcelinna** [comment]
when choosing a domain, several queries come out at once and there is a problem with execution
image
image

**Comment by Marcelinna** [comment]
image

event is added only to button <_CLEAR ALL>, should it also be added to the trash icon next to the service on the bottom bar? is added to the functionality <_Remove from comparison> which does the same as the trash can icon, where there is no functionality.

**Comment by Marcelinna** [comment]
lack of action in 
image

**Comment by Marcelinna** [comment]
lack of action in 
image
image

**Comment by Marcelinna** [comment]
lack of action in 
image

**Comment by Marcelinna** [comment]
lack of request to /probation in Helpdesk-email
image

**Pull request merged**